.github/workflows/yaml-checks.yaml

```
@@ -0,0 +1,30 @@
+name: YAML checks
+
+on:
+  push:
+    branches:
+      - 'main'
+      - '[0-9]+.[1-9][0-9]*.x'
```

I would not restrict to be at least a double-digit number
```suggestion
      - '[0-9]+.[0-9]+.x'
```
Similarly, in the `pull_request` field

---

this doesn't restrict to double digits.
`[1-9]`: one number between 1 and 9
`[0-9]*`: one number between 0 and 9 or no number at all

i set this up a long time ago, cus you can't start versions with a leading 0. So IMO this is correct as it is.